app/src/main/java/org/mifos/mobile/ui/fragments/AccountsFragment.kt

```diff
@@ -363,14 +363,20 @@ class AccountsFragment : BaseFragment(), OnRefreshListener, AccountsView {
      */
     fun filterSavingsAccount(statusModelList: List<CheckboxStatus?>?) {
         val filteredSavings: MutableList<SavingAccount?> = ArrayList()
-        if (accountsPresenter?.getCheckedStatus(statusModelList) != null && accountsPresenter != null) {
+        if (accountsPresenter?.getCheckedStatus(statusModelList) != null && accountsPresenter != null&&accountsPresenter?.getCheckedStatus(statusModelList)?.isNotEmpty()==true) {
             for (status in accountsPresenter?.getCheckedStatus(statusModelList)!!) {
                 accountsPresenter?.getFilteredSavingsAccount(
                     savingAccounts,
                     status,
                 )?.let { filteredSavings.addAll(it) }
             }
         }
+        else if(accountsPresenter?.getCheckedStatus(statusModelList) != null && accountsPresenter != null&& accountsPresenter?.getCheckedStatus(statusModelList)?.isEmpty()==true){
+            savingAccounts?.forEach{
+                filteredSavings.add(it)
+            }
+
+        }
```

Comment:
You could simply do this:
```
   if (accountsPresenter?.getCheckedStatus(statusModelList) != null && accountsPresenter != nul) {
        if (accountsPresenter?.getCheckedStatus(statusModelList)?.isNotEmpty() == true) {
            for (status in accountsPresenter?.getCheckedStatus(statusModelList)!!) {
                accountsPresenter?.getFilteredSavingsAccount(
                    savingAccounts,
                    status,
                )?.let { filteredSavings.addAll(it) }
            }
        } else {
            savingAccounts?.forEach {
                filteredSavings.add(it)
            }
        }
    }
```

And also your **code needs to be reformated**, use Cmd+Option+L (on Mac) or Ctrl+Alt+L (on Windows and Linux)